### DISCUSSION TITLE
AutoRefresh not picking up property changes of Merged Observables

### DISCUSSION THREAD

**Discussion by ruhoch** [reactivemarbles/DynamicData#629] [Q&A]
In my ReactiveDeviceContext class, I am attempting to connect to multiple SourceCaches, Merge the Observables into one, and AutoRefresh the Observable when a particular property changes.

SourceCacheA is a singleton.
SourceCacheB is unique to each instance of ReactiveDeviceContext.
CreateReactiveObserver is only called once per instance of ReactiveDeviceContext

Issues:
1. Some app sessions, changes to the Status property cause the Observable to refresh, other app sessions will not refresh as expected.
2. Furthermore, when I have multiple instances of ReactiveDeviceContext, it seems like the changes are only picked up in the most recently created ReactiveDeviceContext instance. Though I'm not entirely certain this isn't just because the new instance is picking up the current state of the property on initialization.

I expect these issues are due to my own misunderstanding, but the randomness makes it very difficult to figure out what is going wrong here.
```
protected virtual IDisposable CreateReactiveObserver()
{
  IObservable<IChangeSet<ICapability, string>> obsA = this.SourceCacheA.Connect();
  IObservable<IChangeSet<ICapability, string>> obsB = this.SourceCacheB.Connect();

  return obsA
    .Merge(obsB)
    .AutoRefresh(c => c.Status)
    .Filter(c => this.FilterCapability(c))
    .TransformMany(
                    c =>
                    {
                        IEnumerable<ContextBase> contexts = this.ContextBaseResolveMap[c.Name].Select(
                            lazy =>
                            {
                                // initializes ContextBase here...
                            });
                        return contexts;
                    }, c => c)
    .Sort(SortExpressionComparer<ContextBase>.Ascending(c => c.Order))
    .ObserveOn(Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher)
    .Subscribe();
```

**Comment by ruhoch** [comment]
I thought the issue was the Filter, so I eased up the filter to take all changes but I'm still getting this issue. It seemed to help, but I'm still seeing that sometimes changes are ignored.

I've added an event handler to fire for each change to Status, which is firing consistently but still the changes may be ignored in the merged observable.

There is no clear reason why the changes are ignored sometimes and not others, but it seems to either happen across an entire app session or everything works perfectly.

**Comment by ruhoch** [comment]
Our solution was to no longer use this library to observe changes.

**Reply by RolandPheasant** [reply]
I cannot comment on the above snippet as you have not explained what the expectation of Merge is, or what you are trying to achieve by using it.  Merge is an Rx operator and cannot work directly on multiple change sets as changes set have to handle add, updates and removes.  Merge just notifies when one or the other source emits and does not handle the logical  combination of collections.

Instead DD directly supports collection combination operators Or, And and Except to combine multiple collections.
